.github/workflows/build_pytorch_wheels.yml

```diff
@@ -17,16 +20,58 @@ on:
         type: string
 
 permissions:
+  id-token: write
   contents: read
 
 jobs:
   build_pytorch_wheels:
-    name: ${{ inputs.families }} | Python ${{ inputs.python_version }}
-    runs-on: ubuntu-22.04
+    name: Build Linux PyTorch Wheels | ${{ inputs.family }} | Python ${{ inputs.python_version }}
+    runs-on: ${{ github.repository_owner == 'ROCm' && 'azure-linux-scale-rocm' || 'ubuntu-24.04' }}
     container:
       image: ghcr.io/rocm/therock_build_manylinux_x86_64:main
     env:
-      OUTPUT_DIR: /__w/TheRock/TheRock/output
+      OUTPUT_DIR: ${{ github.workspace }}/output
+      PACKAGE_DIST_DIR: ${{ github.workspace }}/output/packages/dist
     steps:
       - name: Checkout
-        uses: actions/checkout@v4
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Configure Git Identity
+        run: |
+          git config --global user.name "therockbot"
+          git config --global user.email "therockbot@amd.com"
+
+      - name: Checkout PyTorch Source Repos
+        run: |
+          echo "Checking out PyTorch repositories..."
+          ./external-builds/pytorch/pytorch_torch_repo.py checkout
+          ./external-builds/pytorch/pytorch_audio_repo.py checkout
+          ./external-builds/pytorch/pytorch_vision_repo.py checkout
+
+      - name: Build PyTorch Wheels
+        run: |
+          echo "Building PyTorch wheels for ${{ inputs.target }}"
+          ./external-builds/pytorch/build_prod_wheels.py build \
```

Comment:
You need to build this in the manylinux docker container. From the script, this is with a command like this:

```
./build_tools/linux_portable_build.py --docker=podman --exec -- \
    /usr/bin/env CCACHE_DIR=/therock/output/ccache \
    /opt/python/cp312-cp312/bin/python \
    /therock/src/external-builds/pytorch/build_prod_wheels.py \
    --pip-cache-dir /therock/output/pip_cache \
    --find-links https://therock-nightly-python.s3.us-east-2.amazonaws.com/gfx110X-dgpu/index.html \
    build \
        --install-rocm \
        --clean \
        --output-dir /therock/output/cp312/wheels \
        --pytorch-rocm-arch gfx1100
```

Note that this also sets the pip-cache-dir to a location that should be valid.

Reply:
The entire workflow is running in the docker container:

```yml
    container:
      image: ghcr.io/rocm/therock_build_manylinux_x86_64:main
```

There are pros and cons to that approach with github actions.

Reply:
Synced online with Stella and we will use the code as is since the workflow running in a docker container. Will use
`--pip-cache-dir /tmp/pipcache` option to avoid permission problem.